{{cookiecutter.project_name}}/contracts/NFT.vy

```diff
@@ -301,6 +317,49 @@ def royaltyInfo(_tokenId: uint256, _salePrice: uint256) -> (address, uint256):
     return self.owner, royalty
 {%- endif %}
 
+{%- if cookiecutter.force_royalties == 'y' %}
+# Helper functions in case market place does not support royalties to execute by this contract with _deductRoyalties()
+@internal
+@view
+def _royaltyInfo(_tokenId: uint256, _salePrice: uint256) -> (address, uint256):
+    """
+    /// @notice Called with the sale price to determine how much royalty
+    //          is owed and to whom. Important; Not all marketplaces respect this, e.g. OpenSea
+    /// @param _tokenId - the NFT asset queried for royalty information
+    /// @param _salePrice - the sale price of the NFT asset specified by _tokenId
+    /// @return receiver - address of who should be sent the royalty payment
+    /// @return owner address and royaltyAmount - the royalty payment amount for _salePrice
+    """
+
+    royalty: uint256 = convert(convert(_salePrice, decimal) * ROYALTY_TO_APPLY_TO_PRICE, uint256) # Percentage that accepts decimals
+    return self.owner, royalty
+
+@internal
+def royaltyChecker(tokenId: uint256):
+     # check if royalties hace been paid
+     if self.balance < self.lastBalance:
+        self._deductRoyalties(tokenId)
+        # equal the contract balance to the lastBalance for future checks
+        self.lastBalance = self.balance
+
+@external
+@payable
+def withdrawRoyalties():
+    assert msg.sender == self.owner
+    amount: uint256 = self.balance
+    send(self.owner, amount)
+    self.lastBalance = 0
+    log RoyaltiesWithdrawn(amount)
+
+@internal
+@payable
+def _deductRoyalties(tokenId: uint256):
+    # we calculate royalties and owners address
+    self.owner, self.royaltyAmount = self._royaltyInfo(tokenId, msg.value)
+    # make transaction to the contract
+    send(self, self.royaltyAmount)
```

Comment:
And then we don't need any of this!
```suggestion
```

Reply:
@fubuloubu would it make sense to add an event to log paid royalties? 
`log RoyaltiesWithdrawn(amount)`

Reply:
You can if you want, the ERC doesn't specify one, and it's only taking a temporary stop in the contract before being pushed to the owner

Reply:
@fubuloubu this one is a bit tricky, because since we `send(self.owner, self.balance) and balance becomes 0, if included before and never makes it to the owner... might need to add a state var only for that... market places that are pro defending artist, might want to check this value to attract more creators? example: "At XXX market place we have distributed to creators XXX royalties".